sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpResponseLogger.java

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.policy;
+
+import com.azure.core.http.HttpResponse;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.logging.LogLevel;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+
+/**
+ * Manages logging HTTP responses in {@link HttpLoggingPolicy}.
+ */
+public interface HttpResponseLogger {
+    /**
+     * Gets the {@link LogLevel} used to log the HTTP response.
+     * <p>
+     * By default this will return {@link LogLevel#INFORMATIONAL}.
+     *
+     * @param response The HTTP response.
+     * @param responseDuration The duration between sending the request and receiving the response.
+     * @return The {@link LogLevel} used to log the HTTP response.
+     */
+    default LogLevel getLogLevel(HttpResponse response, Duration responseDuration) {
```

Comment:
Wondering if `responseDuration` is the only criteria for deciding the log level. For e.g. the level maybe decided on the number of retry attempts too. Can we pass this in a context so we can add other data, if necessary? Same for `logResponse` method too.
`default LogLevel getLogLevel(HttpResponse response, Context context)`

Reply:
Would we want to create a property bag class to pass options into both interfaces, so if we want to expand the information available in the future we don't need to add new APIs? Or do we believe what is being passed now will stay consistent?